README.md

```diff
@@ -1,41 +1,69 @@
+# GeniusYield DEX
+
+This repository houses on-chain smart contracts, Haskell off-chain interaction logic and server endpoints enabling users to easily interact with DEX in language of their choice. 
+
+## Table of Contents
+
+- [Structure of repository](#structure-of-repository)
+- [Spinning up server](#spinning-up-server)
+- [Contributing](#contributing)
+- [License](#license)
+
+## Structure of repository
+
+- [`geniusyield-dex-api`](./geniusyield-dex-api/) provides off-chain code to interact with our DEX. See it's [`README.md`](./geniusyield-dex-api/README.md) for more information about it.
+- [`geniusyield-server-lib`](./geniusyield-server-lib/) serves endpoints using our off-chain code to easily interact with GeniusYield DEX in language of user's choice.
+- [`geniusyield-orderbot`](./geniusyield-orderbot/) cater to additional requirements such as building up of an order-book, receiving price feed, etc.
+
+## Spinning up server
+
+Endpoints made available by server are specified [here](./web/swagger/api.yaml).
+
+1. Make sure your environment is configured properly, consult ["How to build?"](https://atlas-app.io/how-to-build/) section of Atlas documentation for it.
+2. Prepare a configuration, which can be stored either in file or in `SERVER_CONFIG` environment variable. Structure of it is as follows:
+
+    ```yaml
+     # Blockchain provider used by Atlas, our off-chain transaction building tool.
+     # Supported values of `coreProvider` represented as JSON for brevity:
+     # Local node in combination of Kupo, `{ socketPath: string, kupoUrl: string }`
+     # Maestro, `{ maestroToken: string, turboSubmit: boolean }`
+     # Blockfrost, `{ blockfrostKey: string }`
+     # Note that Blockfrost is not recommended as some of the operations performed aren't optimal with it.
+    coreProvider:
+      maestroToken: YOUR_MAESTRO_TOKEN
+      turboSubmit: false
+     # Network id, only `mainnet` and `preprod` are supported for at the moment.
+    networkId: mainnet
+     # Logging configuration. It's an array to cater for potentially multiple scribes.
+    logging:
+      - type:
+           # TODO: Possible values of `tag` are to be documented.
+          tag: stderr
+         # Possible values of `severity` are `Debug`, `Info`, `Warning` and `Error`.
+        severity: Debug
+         # Possible values of `verbosity` are `V0`, `V1`, `V2`, `V3` and `V4`. Consult https://hackage.haskell.org/package/katip-0.8.8.0/docs/Katip.html#t:Verbosity for more information about it.
+        verbosity: V2
+     # Port to serve endpoints at.
+    port: 8082
+     # Maestro API key (token) to access information such as asset details given it's currency symbol and token name.
+    maestroToken: YOUR_MAESTRO_TOKEN
+     # API key to protect server endpoints with. It's value must be provided under `api-key` header of request.
```

Comment:
Such  headers are usually prefixed with x.

In this case I would suggest to use 'x-api-key'.

Reply:
`x-*` prefix is legacy, that's why I decided not to use it.

Reply:
I am nor really sure, if we could say that the `x-` prefix is deprecated in general, but if you prefer to omit it, then we can leave it as it is now.